dataprofiler/tests/data_readers/test_data_utils.py

```diff
@@ -36,3 +37,125 @@ def test_file_UTF_encoding_detection(self):
             detected_encoding = \
                 data_utils.detect_file_encoding(file_path=input_file["path"])
             self.assertEqual(detected_encoding.lower(), input_file["encoding"])
+
+    def test_nth_loc_detection(self):
+        """
+        Tests the ability for the `data_utils.find_nth_location` to detect the
+        nth index of a search_query in a string. 
+        """
+        # Input args: string, query, n
+        # Expected results: index, occurances
+        test_queries = [
+            dict(string="This is a test.", query=".", n=1, index=14, occurances=1),
+            dict(string="This is a test\n", query="\n", n=1, index=14, occurances=1),
+            dict(string="This is a test\nThis is a second test\n",
+                 query="\n", n=0, index=-1, occurances=0),
+            dict(string="This is a test\nThis is a second test\n",
+                 query="\n", n=2, index=36, occurances=2),
+            dict(string='t', query="t", n=1, index=0, occurances=1),
+            dict(string='s', query="t", n=1, index=1, occurances=0),
+            dict(string="This is a test\nThis is a second test\n\n",
+                 query="\n", n=3, index=37, occurances=3),
+            dict(string="This is a test\nThis is a second test\n\nTest\n",
+                 query="\n", n=5, index=43, occurances=4),
+            dict(string="", query="\n", n=3, index=-1, occurances=0),
```

Comment:
ideally i think it would be best if they are broken into dicts of inputs / expected to know exactly via code, but this is fine.
e.g. `dict(input_args=dict(string=.....), expected_results=dict(index=.....)`